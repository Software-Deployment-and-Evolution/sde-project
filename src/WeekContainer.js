import React from 'react';
import DayCard from './DayCard';
import DegreeToggle from './DegreeToggle';

class WeekContainer extends React.Component {
    state = {
        fullData: [],
        dailyData: [],
        degreeType: "fahrenheit"
    }

    componentDidMount = () => {
        const API_KEY = process.env.REACT_APP_WEATHER_API_KEY;
        const weatherURL =
            `http://api.openweathermap.org/data/2.5/forecast?zip=11102&units=imperial&APPID=${API_KEY}`;
        this.fetchingWeatherData(weatherURL);
    }
    fetchingWeatherData = (weatherURL) => {
        fetch(weatherURL)
        .then(res => res.json())
        .then(data => {
            console.log("Data List Loaded", data.list)
            const dailyData = data.list.filter(reading => {
                    return reading.dt_txt.includes("18:00:00")
                }
            )
            this.setState({
                fullData: data.list,
                dailyData: dailyData
            }, () => console.log(this.state))
        });
        
    }
    
    updateForecastDegree = event => {
        this.setState({
            degreeType: event.target.value
        }, () => console.log(this.state))
    }

    formatDayCards = () => {
        return this.state.dailyData.map((reading, index) => <DayCard reading={reading} key={index} degreeType={this.state.degreeType} />)
    }

    render() {
        return (
            <div className="container">
                <h1 className="display-1 jumbotron">4-Day Forecast.</h1>
                <h5 className="display-5 text-muted">New York, US</h5>
                <DegreeToggle degreeType={this.state.degreeType} updateForecastDegree={this.updateForecastDegree} />
                <div className="row justify-content-center">
                    {this.formatDayCards()}
                </div>
            </div>
        )
    }
}

export default WeekContainer;