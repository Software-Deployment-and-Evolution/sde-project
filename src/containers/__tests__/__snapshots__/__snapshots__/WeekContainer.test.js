import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import  WeekContainer from '../../../../WeekContainer'
import DegreeToggle from '../../../../DegreeToggle';

describe('Snapshot Testing: WeekContainer', () => {
  it('renders without crashing with the save snapshot', () => {
    const wrapper = shallow( <WeekContainer /> )
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  it('Snapshot Testing: expect the `degreeType` prop on the DegreeToggle component is the same as default', () => {
    const defaultDegreeType = "farenheit";
    // Query for the DegreeToggle component in the rendered output
    const wrapper = shallow( <WeekContainer /> )
    const DegreeToggleComponent = wrapper.find(DegreeToggle);
    expect(DegreeToggleComponent.props().degreeType).toBe(defaultDegreeType);
  })
})
