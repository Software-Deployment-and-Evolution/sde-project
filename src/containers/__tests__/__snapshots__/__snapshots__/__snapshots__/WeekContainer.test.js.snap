// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing: WeekContainer renders without crashing with the save snapshot 1`] = `
<div
  className="container"
>
  <h1
    className="display-2 jumbotron"
  >
    Weather Forecast
  </h1>
  <h5
    className="display-5"
  >
    Location: New York, US
  </h5>
  <DegreeToggle
    degreeType="farenheit"
    updateForecastDegree={[Function]}
  />
  <div
    className="row"
  />
</div>
`;
